File: CareConnect/CareConnect.Models/SearchObjects/AppointmentAdditionalData.cs
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class AppointmentAdditionalData : BaseAdditionalSearchRequestData
    {
        public bool? IsClientsChildIncluded { get; set; }

        public bool? IsEmployeeAvailabilityIncluded { get; set; }

        public bool? IsAttendanceStatusIncluded { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/search_objects/appointment_additional_data.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment_additional_data.g.dart';

@JsonSerializable(explicitToJson: true)
class AppointmentAdditionalData {
  bool? isClientsChildIncluded;
  bool? isEmployeeAvailabilityIncluded;
  bool? isAttendanceStatusIncluded;

  AppointmentAdditionalData({
    this.isClientsChildIncluded,
    this.isEmployeeAvailabilityIncluded,
    this.isAttendanceStatusIncluded,
  });

  factory AppointmentAdditionalData.fromJson(Map<String, dynamic> json) =>
      _$AppointmentAdditionalDataFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentAdditionalDataToJson(this);
}
